components/service/experiments/src/test/java/mozilla/components/service/experiments/SignatureVerifierTest.kt

```diff
@@ -80,7 +97,16 @@ class SignatureVerifierTest {
     @Test
     fun validSignatureCorrect() {
```

Comment:
This is the single failing test with the current changes.

@pocmo do you remember any context around this test specifically or the signature verification code?

The failure is logged [here](https://tools.taskcluster.net/groups/WJ8WDF4tSMiMcI46LLUBOQ/tasks/ZVybi2lVQSOdkVNSYuDqYQ/runs/0/logs/public%2Flogs%2Flive.log#L6906):
> java.lang.AssertionError: expected:<true> but was:<false>
>   ...
>   at mozilla.components.service.experiments.SignatureVerifierTest.testSignature(SignatureVerifierTest.kt:1022)

Which is the assert [here](https://github.com/mozilla-mobile/android-components/pull/2702/files#diff-f708252b135a5b742cdeb293291a745dR1022):
```kotlin
assertEquals(expected, verifier.validSignature(experiments, experimentsJSON.getLong("last_modified")))
```

The changes in this PR that relates to this are mostly changing the schema of the JSON file (i.e. i changed field names and added/removed some fields), which affects [the test data](https://github.com/mozilla-mobile/android-components/pull/2702/files#diff-f708252b135a5b742cdeb293291a745dR668).
I assume... that changes the signature of the data and i should regenerate it?
Do you know how exactly?

Reply:
Background: Kinto is signing the collection whenever it is changed. On the client side we can verify the signature to make sure the data we received is the correctly signed data and was not altered/intercepted/etc.

For this test code I think we took the actual data/signature from a live Kinto instance used in production, e.g.
* https://firefox.settings.services.mozilla.com/v1/buckets/fennec/collections/experiments/records
* https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/focus-experiments/records

Reply:
@fercarcedo Hey Fernando! Are you still around? I remember you had some great resources about the signing process - do you still have some of those links to share? :)

Reply:
Thanks Sebastian! I found out a bit more: Kinto signs it's experiments using the [autograph service](https://github.com/mozilla-services/autograph), i'm following up on how to get updated signatures out of that.

Reply:
@pocmo Yes, in order to implement it I talked with Franziskus Kiefer and Julien Vehent who pointed me to [this talk](https://www.youtube.com/watch?v=b2kPo8YdLTw&t=0s&list=WL&index=2) and also to  sample implementations of the signature verification algorithm in [C++](https://searchfox.org/mozilla-central/source/security/manager/ssl/ContentSignatureVerifier.cpp) and [Go](https://github.com/mozilla-services/autograph/blob/master/tools/autograph-monitor/contentsignature.go#L35).

Reply:
The test code is using actual data / signature taken from a live Kinto instance. The signature verification code takes the signature and the X5U certificate chain from the Kinto collection metadata URL and validates it (both of which are mocked on the test)

Reply:
Thanks Fernando, that makes sense, i'll go for that.

Reply:
The follow-up will happen in [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1543971).